content/200-concepts/100-components/02-prisma-client/250-opentelemetry-tracing.mdx

```diff
@@ -0,0 +1,324 @@
+
+<Admonition type="info">
+
+Tracing gives you a highly detailed, operation-level insight into your Prisma project. If you want aggregated numerical reporting, such as query counts, connection counts, and total query execution times, see [Metrics](/concepts/components/prisma-client/metrics).
+
+</Admonition>
+
+</TopBlock>
+
+## About tracing
+
+When you enable tracing, Prisma Client outputs the following:
+
+- One trace for each operation (e.g. findMany) that Prisma Client makes.
+- In each trace, one or more [spans](https://www.opentelemetry.io/docs/reference/specification/trace/api/#span). Each span represents the length of time that one stage of the operation takes, such as serialization, or a database query. Spans are represented in a tree structure, where child spans indicate that execution is happening within a larger parent span.
+
+The number and type of spans in a trace varies depending on the type of operation the trace covers, but an example is as follows:
+
+<!-- https://www.figma.com/file/H7EMWAuCaRVHrdNoQpky7J/Tracing?node-id=2%3A37 -->
+
+![image](trace-diagram.png)
+
+You can send tracing output to the console, or analyze it in any OpenTelemetry-compatible tracing system, such as [Jaeger](https://www.jaegertracing.io/), [Honeycomb](https://www.honeycomb.io/trace/) and [Datadog](https://www.datadoghq.com/). On this page, we give an example of how to send tracing output to Jaeger, which you can [run locally](#run-jaeger-locally).
+
+## Prisma's Spans
+
+For each trace, Prisma Client outputs a series of spans. The number and type of these spans varies, depending on the Prisma operation. A typical Prisma trace has the following spans:
+
+- `prisma:client`: Represents the entire Prisma operation, from Prisma Client to the database and back. It contains details such as the model and method called by Prisma Client. Depending on the Prisma operation, it contains one or more of the following spans:
+  - `prisma:engine`: Represents how long a query takes in the [query engine](/concepts/components/prisma-engines/query-engine).
+    - `prisma:engine:connection`: Represents how long it takes for Prisma Client to get a database connection.
+    - `prisma:engine:db_query`: Represents the database query that was executed against the database. It includes the query in the tags, and how long the query took to run.
+
+For example, given the following Prisma Client code:
+
+```ts
+prisma.user.findMany({
+  where: {
+    email: email,
+  },
+  include: {
+    posts: true,
+  },
+})
+```
+
+The trace is structured as follows:
+
+- `prisma:client`
+  - `prisma:engine`:
+    - `prisma:engine:connection`
+    - `prisma:engine:db_query`: details of the first SQL query or command...
+    - `prisma:engine:db_query`: ...details of the next SQL query or command...
+    - `prisma:engine:db_query`: ...and so on, for each SQL query or command
```

Comment:
```suggestion
    - `prisma:engine:serialize`
```

Reply:
Can I check how this works? Is it supposed to be one `prisma:engine:serialize` block after every `prisma:engine:db_query` block, or just one at the end after all of them?

Reply:
Usually, it is one query and one serialize

Reply:
Fixed now